### PR TITLE
Include NullableResponse<T> as an accepted response type

### LINKED ISSUE #4444: AzureSDKAnalyzer: AZC0015 should include NullableResponse<T>
Should also treat NoBodyResponse<T> as deprecated.

### PR DESCRIPTION
**Pull request by christothes**
fixes #4444

This should pass checks once Azure.Core 1.26.0 ships.

### CODE REVIEW

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AzureAnalyzerVerifier.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public static class AzureAnalyzerVerifier<TAnalyzer> where TAnalyzer : Diagnosti
     {
         private static readonly ReferenceAssemblies DefaultReferenceAssemblies =
             ReferenceAssemblies.Default.AddPackages(ImmutableArray.Create(
-                new PackageIdentity("Azure.Core", "1.21.0"),
+                new PackageIdentity("Azure.Core", "1.26.0"),
```
Comment (annelo-msft) [inline src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AzureAnalyzerVerifier.cs:20]: Would it make sense to make this 1.25.0 so we could merge and apply it sooner?

Comment (christothes) [reply src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AzureAnalyzerVerifier.cs:20]: `NullableResponse<T>` needs to exist for the check to work

Comment (annelo-msft) [reply src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AzureAnalyzerVerifier.cs:20]: Got it, makes sense.

**Comment by heaths** [comment]
@christothes when might we see this in Azure/azure-sdk-for-net? Just looked at recent opened and closed PRs and didn't see any corresponding package updates.

**Pull request merged**